### ISSUE TITLE
Possible Silent Refresh Token Issue?

### ISSUE THREAD

**Issue by benhysell** [sounj142/HLSoft.BlazorWebAssembly.Authentication.OpenIdConnect#6]
tldr - testing silent refresh tokens with Identity Server 4 by setting token life times to be very short.
The token appeared to refresh seven times before an exception was thrown.

Details
I wanted to make sure I was correctly configuring my application for silent refresh, and did not want to wait around an hour or so for that workflow to be triggered.  To speed up the process I went into the `dbo.Clients` table in MS SQL and set the following values:

- IdentityTokenLifetime - 30
- AccessTokeLifeTime - 30
- AuthorizationCodeLifetime - 30

Normally these three fields are 300, 3600, and 300.  With timeouts so short this will trigger the silent refresh workflow rather quickly.  And it does, and my app stayed alive for roughly seven refreshes.  On the eight I'll get the following output in the console:
`System.Exception: Exception of type 'System.Exception' was thrown.`

However there are no more details...it dives deep into blazor.webassembly.js rather quickly.

Inside my `App.razor.cs` I hooked up to `SilentRefreshTokenFailEvent`:
```
AuthenticationEventHandler.SilentRefreshTokenFailEvent += (obj, str) =>
            {
                ToastService.ShowError($"Silently refreshing token fail! Error: {str}");
                Console.WriteLine(obj);
            };
```
The `str` and `obj` don't provide any further details either.

I'm not sure if this exception is a library problem...it may be a Blazor wasam issue, however I wasn't sure and thought this was a good place to check first to see if:
1. is there any way to get more details about the exception?
2. Anything else to try to see if it can be narrowed down to the library or Blazor?

**Comment by sounj142** [comment]
Hi benhysell, thanks for your contribution.

I think it looks like in this situation you're doing a "ddos ​​attack" on your development hosting =]]. Oidc-client configures a constant as follows:

const DefaultAccessTokenExpiringNotificationTime = 60; // seconds

This is the number of seconds before the token expires when the client will send a token refresh request, the value here is 60 seconds. Oidc-client also recommends that you should not set the token expiry date lower than it. I think 70s or 80s is suitable number for testing.

In your example, you set the token's lifetime to 30 seconds, which will result the client constantly making token refresh requests. Each request will create an iframe that loads your Blazor app to try a single sign-on. If this is repeated too many times, then you will probably encounter situations that app do not handle in time - I guess this may be the cause the application throwing weird exception you mentioned.

I tested the expiration time setting to 30 seconds, and ran project "Client.IdentityServer.Code.Complex" with both Code and Implicit flows, I didn't see error.

Regarding the obscure exception problem, unfortunately it seems that we don't have a better solution. Blazor handles javascript exceptions by itself so we can only tolerate it, hopefully Blazor will do better in later versions.
